src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/CoverageModelCopyRatioEmissionProbabilityCalculator.java

```diff
@@ -0,0 +1,257 @@
+     */
+    private final int readCountThresholdPoissonSwitch;
+
+    private final boolean applyPoissonToGaussianContinuityCorrection;
+
+    /**
+     * Public constructor.
+     *
+     * @param readCountThresholdPoissonSwitch read count threshold for switching between the exact and inexact
+     *                                        Laplace-approximation-based calculation of the emission probability
+     * @param applyPoissonToGaussianContinuityCorrection if true, continuity correction will be applied
+     */
+    public CoverageModelCopyRatioEmissionProbabilityCalculator(final int readCountThresholdPoissonSwitch,
+                                                               final boolean applyPoissonToGaussianContinuityCorrection) {
+        this.readCountThresholdPoissonSwitch = ParamUtils.isPositiveOrZero(readCountThresholdPoissonSwitch,
+                "The read count threshold for switching to the exact Poisson emission calculation mode (i.e. no" +
+                        " Laplace approximation) must be non-negative");
+        this.applyPoissonToGaussianContinuityCorrection = applyPoissonToGaussianContinuityCorrection;
+    }
+
+    /**
+     * Public constructor.
+     *
+     * @param readCountThresholdPoissonSwitch read count threshold for switching between the exact and inexact
+     *                                        Laplace-approximation-based calculation of the emission probability
+     */
+    public CoverageModelCopyRatioEmissionProbabilityCalculator(final int readCountThresholdPoissonSwitch) {
+        this.readCountThresholdPoissonSwitch = ParamUtils.isPositiveOrZero(readCountThresholdPoissonSwitch,
+                "The read count threshold for switching to the exact Poisson emission calculation mode (i.e. no" +
+                        " Laplace approximation) must be non-negative");
+        this.applyPoissonToGaussianContinuityCorrection = true;
+    }
+
+    /**
+     * Calculate the log emission probability. The parameter {@param target} neglected since
+     * {@param emissionData} contains the necessary information.
+     *
+     * @param emissionData an instance of {@link CoverageModelCopyRatioEmissionData}
+     * @param copyRatio copy ratio on which the emission probability is conditioned on
+     * @param target target on which the emission probability is calculated (this parameter is not used)
+     * @return emission probability
+     */
+    @Override
+    public double logLikelihood(@Nonnull CoverageModelCopyRatioEmissionData emissionData,
+                                double copyRatio,
+                                @Nullable Target target) {
+        Utils.nonNull(emissionData, "The emission data must be non-null");
+        ParamUtils.isPositiveOrZero(copyRatio, "Copy ratio must be non-negative");
+
+        final double logLikelihood;
+        switch (emissionData.getCopyRatioCallingMetadata().getEmissionCalculationStrategy()) {
```

Comment:
fixed. With some changes to the approximations, we don't need to emit `NEGATIVE_INFINITY` anymore. Here's how it reads now:
```
final CopyRatioCallingMetadata.EmissionCalculationStrategy strategy =
        emissionData.getCopyRatioCallingMetadata().getEmissionCalculationStrategy();
final boolean usePoisson = strategy.equals(CopyRatioCallingMetadata.EmissionCalculationStrategy.POISSON) ||
        emissionData.getReadCount() < readCountThresholdPoissonSwitch || copyRatio == 0;
final double logLikelihood = usePoisson ? logLikelihoodPoisson(emissionData, copyRatio)
                                        : logLikelihoodLaplaceApproximation(emissionData, copyRatio);